File: 07-steroid_langs/1-scala/src/main/scala/codely/contexts/rrss/users/application/find/UserFinder.scala
package codely.contexts.rrss.users.application.find

import codely.contexts.rrss.users.domain.{UserDoesNotExistError, UserRepository, User}

class UserFinder(repository: UserRepository) {
	def find(id: String): Either[UserDoesNotExistError, User] = {
		repository.search(id) match {
			case None => Left(UserDoesNotExistError(id))
			case Some(user) => Right(user)
		}
	}
}

File: 07-langs_particularities/1-php/src/Contexts/Users/Application/Find/UserFinder.php
<?php
declare(strict_types=1);

namespace CodelyTv\Contexts\Users\Application\Find;

use CodelyTv\Contexts\Users\Domain\User;
use CodelyTv\Contexts\Users\Domain\UserDoesNotExistError;
use CodelyTv\Contexts\Users\Domain\UserRepository;

final class UserFinder
{
	public function __construct(private readonly UserRepository $repository)
	{
	}

	/** @throws UserDoesNotExistError */
	public function find(string $id): User
	{
		$post = $this->repository->search($id);

		if (null === $post) {
			throw new UserDoesNotExistError($id);
		}

		return $post;
	}
}
